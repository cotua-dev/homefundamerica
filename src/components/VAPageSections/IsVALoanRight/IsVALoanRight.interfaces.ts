export interface FetchIsVALoanRightImageQuery {
    isVALoanRightImage: {
        publicURL: string;
    };
}
