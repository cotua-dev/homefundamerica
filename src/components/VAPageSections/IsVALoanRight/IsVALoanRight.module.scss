@import '../../../styles/variables';

section#isVALoanRightSection {
    display: flex;
    align-items: center;
    justify-content: space-evenly;
    flex-direction: row;
    color: $primaryColor;
    height: 60vh;

    @media screen and (max-width: 1000px) {
        flex-direction: column;
        height: 100%;
        padding: 40px 10px;
    }

    div.imageWrapper {
        img.isVALoanRightImage {
            width: 500px;
            height: auto;
            border-radius: 10px;

            @media screen and (max-width: 600px) {
                width: 100%;
            }
        }
    }

    div.contentWrapper {
        width: 500px;
        display: flex;
        flex-direction: column;
        justify-content: center;

        @media screen and (max-width: 500px) {
            width: 100%;
        }

        h2.title {
            font-size: 32px;

            @media screen and (max-width: 950px) {
                font-size: 24px;
            }
        }

        p.content,
        ul.list li.listItem {
            font-size: 16px;

            @media screen and (max-width: 950px) {
                font-size: 14px;
            }
        }

        ul.list {
            padding-left: 20px;
        }

        button.applyButton {
            border: 0;
            border-radius: 10px;
            color: $textColor;
            font-size: 18px;
            background-color: $secondaryColor;
            padding: 15px 20px;
            margin-top: 20px;
            cursor: pointer;
            transition: 0.8s;

            &:hover {
                background-color: $primaryColor;
            }

            @media screen and (max-width: 1000px) {
                margin-bottom: 20px;
                width: 100%;
            }
        }
    }
}
