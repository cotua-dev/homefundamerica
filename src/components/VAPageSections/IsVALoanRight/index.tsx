import React, {
    FunctionComponent,
    ReactElement,
} from 'react';
import {
    useStaticQuery,
    graphql,
} from 'gatsby';
import { FetchIsVALoanRightImageQuery } from './IsVALoanRight.interfaces';
import * as styles from './IsVALoanRight.module.scss';

export const IsVALoanRight: FunctionComponent = (): ReactElement => {
    const queryResult = useStaticQuery<FetchIsVALoanRightImageQuery>(graphql`
        query {
            isVALoanRightImage: file(name: {eq: "AdobeStock_107429959"}, extension: {eq: "jpg"}) {
                publicURL
            }
        }
    `);

    return (
        <section id={styles.isVALoanRightSection}>
            <div className={styles.imageWrapper}>
                <img
                    className={styles.isVALoanRightImage}
                    src={queryResult?.isVALoanRightImage.publicURL}
                    alt="Is a VA loan right image"
                />
            </div>
            <div className={styles.contentWrapper}>
                <h2 className={styles.title}>{`Is a VA Loan Right for You?`}</h2>
                <p className={styles.content}>{`A VA loan may be right for you if you:`}</p>
                <ul className={styles.list}>
                    <li className={styles.listItem}>
                        {`
                            Veteran who meets the eligibility requirements
                            established by the U.S Department of Veterans
                            Affairs*
                        `}
                    </li>
                    <li className={styles.listItem}>{`Want the option of making no down payment`}</li>
                    <li className={styles.listItem}>
                        {`
                            Want to save money by avoiding monthly Private
                            Mortgage Insurance (PMI) requirements
                        `}
                    </li>
                </ul>
            </div>
        </section>
    );
}
