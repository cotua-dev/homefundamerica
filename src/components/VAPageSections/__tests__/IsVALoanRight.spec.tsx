import React from 'react';
import {
    render,
    RenderResult,
} from '@testing-library/react';
import { IsVALoanRight } from '../IsVALoanRight';

describe('IsVALoanRight Component', (): void => {
    test('renders', (): void => {
        const { container }: RenderResult = render(<IsVALoanRight/>);
        expect(container.firstChild).toMatchSnapshot();
    });
});
