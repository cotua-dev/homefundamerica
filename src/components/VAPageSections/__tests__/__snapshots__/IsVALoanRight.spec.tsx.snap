// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IsVALoanRight Component renders 1`] = `
<section>
  <div>
    <img
      alt="Is a VA loan right image"
    />
  </div>
  <div>
    <h2>
      Is a VA Loan Right for You?
    </h2>
    <p>
      A VA loan may be right for you if you:
    </p>
    <ul>
      <li>
        
                            Veteran who meets the eligibility requirements
                            established by the U.S Department of Veterans
                            Affairs*
                        
      </li>
      <li>
        Want the option of making no down payment
      </li>
      <li>
        
                            Want to save money by avoiding monthly Private
                            Mortgage Insurance (PMI) requirements
                        
      </li>
    </ul>
  </div>
</section>
`;
